Code/Framework/AzFramework/AzFramework/Terrain/TerrainDataRequestBus.h

```diff
@@ -59,6 +59,10 @@ namespace AzFramework
                 const AZ::Vector2& position, Sampler sampler = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const = 0;
             virtual float GetHeightFromFloats(
                 float x, float y, Sampler sampler = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const = 0;
+            //! These functions are required for python tests as we can't pass the terrainExistsPtr.
+            virtual float GetHeightOnly(const AZ::Vector3& position, Sampler sampler = Sampler::BILINEAR) const = 0;
+            virtual float GetHeightFromVector2Only(const AZ::Vector2& position, Sampler sampler = Sampler::BILINEAR) const = 0;
+            virtual float GetHeightFromFloatsOnly(float x, float y, Sampler sampler = Sampler::BILINEAR) const = 0;
```

Comment:
Please don't expand the public API just for python purposes.  The pattern we're establishing is to create private non-virtual methods that call through to the public API.  In this case, it would look like this:
```
private:
float BehaviorContextGetHeight(const AZ::Vector3& position, Sampler sampler = Sampler::BILINEAR)
{
    return GetHeight(position, sampler);
}
```
See the recently-added BehaviorContextGetSurfacePoint methods as an example.  This way we don't need to add mocks, more implementations, etc.

Reply:
All changed.